panel/layout/modal.py

```diff
@@ -0,0 +1,187 @@
+  background-color: var(--background-color);
+  border-radius:  calc(var(--corner-radius) * 1px);
+}
+@keyframes fade-in {
+  from {
+    opacity: 0;
+  }
+}
+@keyframes slide-up {
+  from {
+    transform: translateY(10%);
+  }
+}
+.dialog-overlay {
+  animation: fade-in 200ms both;
+}
+.dialog-content {
+  animation: fade-in 400ms 200ms both, slide-up 400ms 200ms both;
+}
+@media (prefers-reduced-motion: reduce) {
+  .dialog-overlay,
+  .dialog-content {
+    animation: none;
+  }
+}
+.pnx-dialog-close {
+  position: absolute;
+  top: 0.5em;
+  right: 0.5em;
+  border: 0;
+  padding: 0.25em;
+  background-color: transparent;
+  font-size: 1.5em;
+  width: 1.5em;
+  height: 1.5em;
+  text-align: center;
+  cursor: pointer;
+  transition: 0.15s;
+  border-radius: 50%;
+  z-index: 10003;
+}
+fast-design-system-provider .pnx-dialog-close {
+  color:  var(--neutral-foreground-rest);
+}
+.pnx-dialog-close:hover {
+  background-color: rgb(50 50 0 / 0.15);;
+}
+fast-design-system-provider .pnx-dialog-close:hover {
+  background-color: var(--neutral-fill-hover);
+}
+.lm-Widget.p-Widget.lm-TabBar.p-TabBar.lm-DockPanel-tabBar.jp-Activity {
+  z-index: -1;
+}
+"""
+
+class Modal(ReactiveHTML, NamedListLike):  # pylint: disable=too-many-ancestors
+    """The `Modal` layout is a *pop up* element that displays in front of and deactivates
+    all other page content.
+
+    You will need to include the `Modal` in a layout or your template. It will not be
+    shown before you trigger the `open` event or equivalently set `is_open=True`.
+
+    Args:
+        *objects: The objects to display in the modal.
+
+    Reference: https://panel.holoviz.org/reference/layouts/Modal.html
+
+    Example:
+
+    >>> import panel as pn
+    >>> from panel import Modal
+    >>> pn.extension("modal")
+    >>> modal = Modal(pn.panel("Hi. I am the Panel Modal!", width=200))
+    >>> pn.Column(modal.param.open, modal).servable()
+    """
+
+    is_open = param.Boolean(doc="""
+        Whether or not the modal is open. Set to True to open. Set to False to close.""")
+    show_close_button = param.Boolean(True, doc="Whether to show a close button in the modal")
+
+    open = param.Event(doc="Click here to open the modal")
+    close = param.Event(doc="Click here to close the modal")
+
+    style = param.String(STYLE, doc="The css styles applied to the modal")
+
+    def __init__(self, *objects, **params):  # pylint: disable=redefined-builtin
+        params["height"] = params["width"] = params["margin"] = 0
+
+        NamedListLike.__init__(self, *objects, **params)
+        ReactiveHTML.__init__(self, objects=self.objects, **params)
+
+    @param.depends("open", watch=True)
+    def _show(self):
+        self.is_open = True
+
+    @param.depends("close", watch=True)
+    def _hide(self):
+        self.is_open = False
+
+    _extension_name = "modal"
+
+    __javascript__ = [JS_FILE]
+
+
+    _template = """
+<style id="pnx_dialog_style">
```

Comment:
I should maybe find another way to include the style. The problem is that with the fast templates this style needs to be available globally while with other templates it does not have to.